Sources/ScipioKit/Producer/PIF/PIFGenerator.swift

```diff
@@ -1,308 +1,208 @@
 import Foundation
 import Basics
+import PIFKit
 import SPMBuildCore
 import PackageModel
 import PackageGraph
 import XCBuildSupport
 
-extension PIF.Target {
-    fileprivate enum TargetType {
-        case library
-        case resourceBundle
-    }
-
-    fileprivate var supportedType: TargetType? {
-        switch productType {
-        case .objectFile:
-            return .library
-        case .bundle:
-            return .resourceBundle
-        default: // unsupported types
-            return nil
-        }
-    }
-}
-
 struct PIFGenerator {
-    private let descriptionPackage: DescriptionPackage
+    private let packageName: String
+    private let packageLocator: any PackageLocator
     private let buildParameters: BuildParameters
     private let buildOptions: BuildOptions
     private let buildOptionsMatrix: [String: BuildOptions]
+    private let executor: any Executor
     private let fileSystem: any FileSystem
 
     init(
-        package: DescriptionPackage,
+        packageName: String,
+        packageLocator: some PackageLocator,
         buildParameters: BuildParameters,
```

Comment:
Because we don't use `PIFBuilder.generatePIF` anymore, only `buildParameters.toolchain.toolchainLibDir` is needed. How about passing the specific value instead of the entire `BuildParameters`?